libraries/Microsoft.Bot.Builder.AI.QnA/Helpers/GenerateAnswerHelper.cs

```diff
@@ -0,0 +1,193 @@
+                }
+
+                if (queryOptions.Top != hydratedOptions.Top && queryOptions.Top != 0)
+                {
+                    hydratedOptions.Top = queryOptions.Top;
+                }
+
+                if (queryOptions.StrictFilters?.Length > 0)
+                {
+                    hydratedOptions.StrictFilters = queryOptions.StrictFilters;
+                }
+
+                if (queryOptions.MetadataBoost?.Length > 0)
+                {
+                    hydratedOptions.MetadataBoost = queryOptions.MetadataBoost;
+                }
+            }
+
+            return hydratedOptions;
+        }
+
+        private void ValidateOptions(QnAMakerOptions options)
+        {
+            if (options.ScoreThreshold == 0)
+            {
+                options.ScoreThreshold = 0.3F;
+            }
+
+            if (options.Top == 0)
+            {
+                options.Top = 1;
+            }
+
+            if (options.ScoreThreshold < 0 || options.ScoreThreshold > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(options.ScoreThreshold), "Score threshold should be a value between 0 and 1");
+            }
+
+            if (options.Timeout == 0.0D)
+            {
+                options.Timeout = 100000;
+            }
+
+            if (options.Top < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(options.Top), "Top should be an integer greater than 0");
+            }
+
+            if (options.StrictFilters == null)
+            {
+                options.StrictFilters = new Metadata[] { };
+            }
+
+            if (options.MetadataBoost == null)
+            {
+                options.MetadataBoost = new Metadata[] { };
+            }
+        }
+
+        private async Task<QueryResult[]> QueryQnaServiceAsync(Activity messageActivity, QnAMakerOptions options)
+        {
+            var requestUrl = $"{_endpoint.Host}/knowledgebases/{_endpoint.KnowledgeBaseId}/generateanswer";
+            var jsonRequest = JsonConvert.SerializeObject(
+                new
+                {
+                    question = messageActivity.Text,
+                    top = options.Top,
+                    strictFilters = options.StrictFilters,
+                    metadataBoost = options.MetadataBoost,
+                    scoreThreshold = options.ScoreThreshold,
+                }, Formatting.None);
+
+            var httpRequestHelper = new HttpRequestHelper(httpClient);
+            var response = await httpRequestHelper.ExecuteHttpRequest(requestUrl, jsonRequest, _endpoint).ConfigureAwait(false);
+
+            var result = await FormatQnaResultAsync(response, options).ConfigureAwait(false);
+
+            return result;
+        }
+
+        private async Task EmitTraceInfoAsync(ITurnContext turnContext, Activity messageActivity, QueryResult[] result, QnAMakerOptions options)
```

Comment:
static

Reply:
Can't make this function static as it has dependency on member variable (_endpoint) of this class.